spring-rest2ts-jackson-js/src/main/java/com/blueveery/springrest2ts/converters/JacksonAnnotationsConversionToJacksonJs.java

```diff
@@ -0,0 +1,208 @@
+        }
+    }
+
+    private String getMinimalClassName(Class javaType, Class javaRoot) {
+        if ( javaType.getName().startsWith(javaRoot.getPackage().getName())) {
+            return javaType.getName().replaceFirst(javaRoot.getPackage().getName(), "");
+        }
+        return javaType.getName();
+    }
+
+    private Class findInheritanceRoot(Class javaType) {
+        Class superclass = javaType.getSuperclass();
+        if (superclass != null && superclass != Object.class && TypeMapper.map(superclass) != TypeMapper.tsAny) {
+            if (superclass.getAnnotation(JsonTypeInfo.class) == null) {
+                return findInheritanceRoot(superclass);
+            } else {
+                return superclass;
+            }
+        }
+        return null;
+    }
+
+    private TSDecorator addJsonSubTypesAnnotation(TSClass tsClass) {
+        TSDecorator jsonSubTypesDecorator = new TSDecorator(jsonSubTypesFunction);
+        TSJsonLiteral literal = new TSJsonLiteral();
+        literal.getFieldMap().put("types", new TSLiteralArray());
+        jsonSubTypesDecorator.getTsLiteralList().add(literal);
+        tsClass.getTsDecoratorList().add(jsonSubTypesDecorator);
+        tsClass.addScopedTypeUsage(jsonSubTypesFunction);
+        return jsonSubTypesDecorator;
+    }
+
+    private void addJsonTypeInfoAnnotation(TSClass tsClass, JsonTypeInfo jsonTypeInfo) {
+        TSDecorator jsonTypeInfoDecorator = new TSDecorator(jsonTypeInfoFunction);
+        TSJsonLiteral jsonTypeInfoLiteral = new TSJsonLiteral();
+        jsonTypeInfoDecorator.getTsLiteralList().add(jsonTypeInfoLiteral);
+        tsClass.addScopedTypeUsage(jsonTypeInfoFunction);
+
+        convertJsonTypeInfoUse(tsClass, jsonTypeInfo, jsonTypeInfoLiteral);
+        if (convertJsonTypeInfoInclude(tsClass, jsonTypeInfo, jsonTypeInfoLiteral)) {
+            convertJsonTypeInfoProperty(jsonTypeInfo, jsonTypeInfoLiteral);
+        }
+
+        tsClass.getTsDecoratorList().add(0, jsonTypeInfoDecorator);
+    }
+
+    private void convertJsonTypeInfoProperty(JsonTypeInfo jsonTypeInfo, TSJsonLiteral jsonTypeInfoLiteral) {
+        String propertyName = jsonTypeInfo.property();
+        if("".equals(propertyName)) {
+            propertyName = jsonTypeInfo.use().getDefaultPropertyName();
+        }
+        if (!"@type".equals(propertyName)) {
+            jsonTypeInfoLiteral.getFieldMap().put("property", new TSLiteral("", TypeMapper.tsString, propertyName));
+        }
+    }
+
+    private void convertJsonTypeInfoUse(TSClass tsClass, JsonTypeInfo jsonTypeInfo, TSJsonLiteral jsonTypeInfoLiteral) {
+        if (Arrays.asList(JsonTypeInfo.Id.NAME, JsonTypeInfo.Id.CLASS, JsonTypeInfo.Id.MINIMAL_CLASS).contains(jsonTypeInfo.use())) {
+            jsonTypeInfoLiteral.getFieldMap().put("use", new TSLiteral("", TypeMapper.tsAny, "JsonTypeInfoId.NAME"));
+            tsClass.addScopedTypeUsage(jsonTypeInfoIdEnum);
+        } else {
```

Comment:
``else``  statement is not required if you throw an exception

Reply:
if I add return statement in IF branch but this is not readable for me